[Ai Ling] iP

ailing35 opened the pull request:
### Abyss
> "The Abyss is a dark place." - Ai Ling

Abyss frees your mind of having to remember things you need to do. It's,
* text-based
* easy to learn
* ~~FAST~~ _SUPER_ FAST to use

All you need to do is,
1. download it from [here](https://github.com/ailing35/ip/releases/download/A-Jar/ip.jar).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉

And it is __FREE__!

Features:

- [x] Managing tasks
- [x] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)

If you are a Java programmer, you can use it to practice Java too. Here's the```main``` method:

```java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

ailing35 committed:
Add greet, echo and exit features

ailing35 committed:
Add listing and adding to list features

ailing35 committed:
Add mark as done feature

ailing35 committed:
Add todo, event and deadline task features

ailing35 committed:
Add automated Text UI Testing

ailing35 committed:
Add error handling

ailing35 committed:
Add delete feature

ailing35 committed:
Add exception subclasses

ailing35 committed:
Add save tasks feature

ailing35 committed:
Use date type and format date

ailing35 committed:
Merge branch 'branch-Level-8'

ailing35 committed:
Abstract out storage features

ailing35 committed:
Merge branch 'branch-Level-7'

ailing35 committed:
Parse date string in storage

ailing35 committed:
Extend OOP design

ailing35 committed:
Organise into packages

ailing35 committed:
Add JUnit tests and refactor exceptions

ailing35 committed:
Add JavaDoc comments

ailing35 committed:
Tweak code to comply with coding standard

ailing35 committed:
Add find task feature

ailing35 committed:
Merge branch 'branch-Level-9'

ailing35 committed:
Merge branch 'branch-A-CodingStandard'

ailing35 committed:
Merge branch 'branch-A-JavaDoc'

ailing35 committed:
Clean up import statements and add javadoc comments

ailing35 committed:
Update README.md

ailing35 committed:
Merge remote-tracking branch 'origin/add-gradle-support'

adam-ky commented on src/main/java/abyss/command/Parser.java:
```diff
@@ -0,0 +1,46 @@
+            content = parts[1];
+        }
+
+        switch (type) {
```
I like how neat everything inside the switch block is 👍

adam-ky commented on src/main/java/abyss/Ui.java:
```diff
@@ -0,0 +1,81 @@
+     * @return Formatted reply.
+     */
+    public static String formatReply(String[] messages) {
+        String reply = "\t......................................................\n";
```
Perhaps you can declare a final member variable to represent this line?

adam-ky commented on src/main/java/abyss/Abyss.java:
```diff
@@ -0,0 +1,94 @@
+        }
+    }
+
+    public static int getNumberOfTasks() {
```
Do remember to include JavaDoc comments for the remaining methods, once you have the time, of course 😄 💪

adam-ky reviewed (commented):
Coding standards are well adhered to!

ngsengleng commented on src/main/java/abyss/task/TaskList.java:
```diff
@@ -46,6 +46,20 @@ public void markAsDone(int i) {
 
+    public void find(String keyword) {
+        keyword = keyword.trim();
+        String regex = "[ -~]*" + keyword + "[ -~]*";
```
Nice use of regex! 👍

ngsengleng commented on src/main/java/abyss/task/Event.java:
```diff
@@ -10,7 +10,7 @@
     protected LocalDate at;
```
maybe this variable name can be more descriptive?

ngsengleng commented on src/main/java/abyss/exception/AbyssException.java:
```diff
@@ -1,6 +1,14 @@
+ * Represents an exception that occurs in Abyss.
+ */
 public class AbyssException extends Exception {
+    /**
+     * Constructor for AbyssException.
+     *
+     * @param message Error message.
+     */
```
Perhaps javadoc comments for constructors are not needed since they are not methods?

ngsengleng commented on src/main/java/abyss/task/TaskList.java:
```diff
@@ -12,28 +12,58 @@
+     */
     public void delete(int i) {
         String removedMsg = "Task piece is swallowed by the Abyss.";
         String tasksLeftMsg = "The Abyss now contains " + (getNumberOfTasks() - 1) + " task piece(s).";
```
String.format can be another alternative and may look much cleaner 😃

ngsengleng reviewed (commented):
Overall code quality is great! Very well structured classes and methods and code mostly if not fully adheres to java coding standards. Nice job 👍

ailing35 committed:
Add checkStyle and fix code style

ailing35 committed:
Add GUI to Abyss with FXML

ailing35 committed:
Merge branch 'branch-Level-10' into master

ailing35 committed:
Update gradle build configurations

ailing35 committed:
Add assert statements

Assert statements are used to document assumptions of non-empty task
descriptions that should hold.

ailing35 committed:
Refactor the execution of commands

The execution of commands is refactored to the individual command types.
This is to make use of polymorphism to eliminate the need for command
type check, improving the readability of code.

ailing35 committed:
Merge pull request #1 from ailing35/branch-A-Assertions

Add assert statements

ailing35 committed:
Resolve merge conflict

ailing35 committed:
Merge pull request #2 from ailing35/branch-A-CodeQuality

Refactor the execution of commands

ailing35 committed:
Add update task feature

ailing35 committed:
Merge branch 'branch-C-Update' into master

ailing35 committed:
Improve user interface

ailing35 committed:
Merge branch 'branch-A-BetterGui' into master

ailing35 committed:
Update README and add product screenshot

ailing35 committed:
Set theme jekyll-theme-slate

ailing35 committed:
Set theme jekyll-theme-midnight

ailing35 committed:
Update README.md

ailing35 committed:
Make find case-insensitive

ailing35 committed:
Merge branch 'master' of github.com:ailing35/ip into master

ailing35 committed:
Update README.md

ailing35 committed:
Update README.md

ailing35 committed:
Fix jar file

ailing35 committed:
Merge branch 'master' of github.com:ailing35/ip into master

ailing35 committed:
Reduce exit delay
